addons/web/static/src/js/chrome/web_client.js

```diff
@@ -74,36 +74,31 @@ return AbstractWebClient.extend({
                 return menuData;
             });
     },
-    show_application: function () {
-        var self = this;
+    async show_application() {
         this.set_title();
 
-        return this.menu_dp.add(this.instanciate_menu_widgets()).then(function () {
-            $(window).bind('hashchange', self.on_hashchange);
-
-            // If the url's state is empty, we execute the user's home action if there is one (we
-            // show the first app if not)
-            var state = $.bbq.getState(true);
-            if (_.keys(state).length === 1 && _.keys(state)[0] === "cids") {
-                return self.menu_dp.add(self._rpc({
-                        model: 'res.users',
-                        method: 'read',
-                        args: [session.uid, ["action_id"]],
-                    }))
-                    .then(function (result) {
-                        var data = result[0];
-                        if (data.action_id) {
-                            return self.do_action(data.action_id[0]).then(function () {
-                                self.menu.change_menu_section(self.menu.action_id_to_primary_menu_id(data.action_id[0]));
-                            });
-                        } else {
-                            self.menu.openFirstApp();
-                        }
-                    });
-            } else {
-                return self.on_hashchange();
-            }
-        });
+        await this.menu_dp.add(this.instanciate_menu_widgets());
+        $(window).bind('hashchange', this.on_hashchange);
+
+        const state = $.bbq.getState(true);
+        if (!_.isEqual(_.keys(state), ["cids"])) {
```

Comment:
I don't know the API of underscore well enough (probably because I would prefer to remove it completely and never uses it) but isn't this the same as `!Object.keys(state).includes("cids") ?

Reply:
It would be more something like Object.keys(state) == ["cids"] if it was possible, include is not enough.

Maybe `!_.isEqual(Object.keys(state), ["cids"])` but still using undescore...

Or `JSON.stringify(Object.keys(state)) == '["cids"]'` 💩

Reply:
> I don't know the API of underscore well enough (probably because I would prefer to remove it completely and never uses it) but isn't this the same as `!Object.keys(state).includes("cids")` ?

No, `includes` would just check if there's a `cids` key, `isEqual` checks that `cids` is the one key. That's what the original code did "by hand" so I replicated this behaviour.

Reply:
Thanks for the clarificaton. I don't really care what the final version of this check looks like, I just don't like underscore. Feel free to keep it the way it is.